### PR TITLE
Remove TODOs (moved to issues)

### LINKED ISSUE #95: Remove all `todo()`s from the codebase
After adding error handling, a few calls to `todo()` still remain; remove them.

### PR DESCRIPTION
**Pull request by JulianVentura**
This PR removes TODO comments in code, since they where moved to issues.

Closes #95

### CODE REVIEW

**Review by jrchatruc — APPROVED** [review]

**Review by gianbelinche — APPROVED** [review]
LGTM
nit: maybe add comments on this pr for each todo, stating what issue was created for it or if it was deleted, mainly to make sure we don't forget any

File: docs/general.md [hunk 1]
```diff
@@ -259,5 +259,4 @@ add 8,r0,r1
 
 ## Bootloader
 
-TODO
```
Comment (JulianVentura) [inline docs/general.md:262]: Docs TODOs handled on hackmd document.

File: src/lib.rs [hunk 2]
```diff
@@ -116,7 +116,7 @@ pub fn run_program(
 ) -> ExecutionOutput {
     match run_program_with_error(bin_path, vm, storage, tracers) {
         Ok((execution_output, _vm)) => execution_output,
-        Err(_) => ExecutionOutput::Panic, // TODO: fix this
```
Comment (JulianVentura) [inline src/lib.rs:119]: Created issue #117

File: src/lib.rs [hunk 3]
```diff
@@ -203,9 +203,6 @@ pub fn run(
                 Variant::FarCall(far_call_variant) => {
                     far_call(&mut vm, &opcode, &far_call_variant, storage)
                 }
-                // TODO: This is not how return works. Fix when we have calls between contracts
```
Comment (JulianVentura) [inline src/lib.rs:206]: This is an old TODO

File: src/op_handlers/context.rs [hunk 4]
```diff
@@ -45,10 +45,10 @@ pub fn meta(vm: &mut VMState, opcode: &Opcode) -> Result<(), EraVmError> {
                 .get(vm.current_frame()?.aux_heap_id)
                 .ok_or(HeapError::ReadOutOfBounds)?
                 .len() as u32,
-            this_shard_id: 0,   //
-            caller_shard_id: 0, // TODO: shard_id related stuff is not implemented yet
-            code_shard_id: 0,   //
-            aux_field_0: 0,     // TODO: this should only be zero when not in kernel mode
```
Comment (JulianVentura) [inline src/op_handlers/context.rs:51]: Created issue #112

File: src/op_handlers/context.rs [hunk 5]
```diff
@@ -45,10 +45,10 @@ pub fn meta(vm: &mut VMState, opcode: &Opcode) -> Result<(), EraVmError> {
                 .get(vm.current_frame()?.aux_heap_id)
                 .ok_or(HeapError::ReadOutOfBounds)?
                 .len() as u32,
-            this_shard_id: 0,   //
-            caller_shard_id: 0, // TODO: shard_id related stuff is not implemented yet
```
Comment (JulianVentura) [inline src/op_handlers/context.rs:49]: Deleted, since currently this is not a feature on era

File: src/op_handlers/context.rs [hunk 6]
```diff
@@ -77,9 +77,6 @@ pub fn set_context_u128(vm: &mut VMState, opcode: &Opcode) -> Result<(), EraVmEr
 }
 
 pub fn aux_mutating0(_vm: &mut VMState, _opcode: &Opcode) -> Result<(), EraVmError> {
-    // None of the other vms have the implementation for this opcode
-    // from spec document:
-    // (* TODO: the following is not implemented yet *)
```
Comment (JulianVentura) [inline src/op_handlers/context.rs:82]: Created issue #124

File: src/op_handlers/event.rs [hunk 7]
```diff
@@ -15,7 +15,7 @@ pub fn event(vm: &mut VMState, opcode: &Opcode) -> Result<(), EraVmError> {
             key,
             value,
             is_first: opcode.alters_vm_flags,
-            shard_id: 1, // TODO: Shard Ids are not yet implemented
```
Comment (JulianVentura) [inline src/op_handlers/event.rs:18]: Deleted, since currently this is not a feature on era

File: src/op_handlers/far_call.rs [hunk 8]
```diff
@@ -136,11 +136,6 @@ pub fn far_call(
     far_call: &FarCallOpcode,
     storage: &mut dyn Storage,
 ) -> Result<(), EraVmError> {
-    /*
-        TODO:
```
Comment (JulianVentura) [inline src/op_handlers/far_call.rs:140]: Deleted, since it has already been resolved

File: src/op_handlers/far_call.rs [hunk 9]
```diff
@@ -205,7 +200,6 @@ pub fn far_call(
 
             vm.clear_flags();
 
-            // TODO: EVM interpreter stuff.
```
Comment (JulianVentura) [inline src/op_handlers/far_call.rs:208]: Created issue #123

File: src/op_handlers/far_call.rs [hunk 10]
```diff
@@ -256,7 +250,6 @@ pub fn far_call(
...
```
Comment (JulianVentura) [inline src/op_handlers/far_call.rs:259]: Created issue #123

File: src/op_handlers/heap_read.rs [hunk 11]
```diff
@@ -21,7 +21,7 @@ pub fn heap_read(vm: &mut VMState, opcode: &Opcode) -> Result<(), EraVmError> {
         .heaps
         .get_mut(vm.current_frame()?.heap_id)
         .ok_or(HeapError::StoreOutOfBounds)?
-        .expand_memory(addr + 32); // TODO: Handle ergs cost
```
Comment (JulianVentura) [inline src/op_handlers/heap_read.rs:24]: Created issue #110

File: src/op_handlers/near_call.rs [hunk 12]
```diff
@@ -7,7 +7,7 @@ use crate::{opcode::Opcode, state::VMState};
 pub fn near_call(vm: &mut VMState, opcode: &Opcode) -> Result<(), EraVmError> {
     let abi_reg = vm.get_register(opcode.src0_index);
     let callee_address = opcode.imm0;
-    let exception_handler = opcode.imm1; //TODO: Add exception handler to call frame
```
Comment (JulianVentura) [inline src/op_handlers/near_call.rs:10]: Deleted, since it has already been resolved.

File: src/op_handlers/precompile_call.rs [hunk 13]
```diff
@@ -18,8 +18,6 @@ use zkevm_opcode_defs::{
 use crate::{eravm_error::EraVmError, heaps::Heaps, state::VMState, value::TaggedValue, Opcode};
 
 pub fn precompile_call(vm: &mut VMState, opcode: &Opcode) -> Result<(), EraVmError> {
-    // TODO check that we're in a system call
```
Comment (JulianVentura) [inline src/op_handlers/precompile_call.rs:21]: Created issue #112

File: src/opcode.rs [hunk 14]
```diff
@@ -24,7 +24,6 @@ impl From<u8> for Predicate {
             x if x == Predicate::Le as u8 => Predicate::Le,
             x if x == Predicate::Ne as u8 => Predicate::Ne,
             x if x == Predicate::GtOrLt as u8 => Predicate::GtOrLt,
-            // TODO: maybe don't panic here
```
Comment (JulianVentura) [inline src/opcode.rs:27]: Created issue #114

File: src/state.rs [hunk 15]
```diff
@@ -238,7 +237,7 @@ impl VMState {
         }
     }
 
-    #[allow(clippy::too_many_arguments)] // TODO: check if we can avoid this
```
Comment (JulianVentura) [inline src/state.rs:241]: Created issue #115

File: src/state.rs [hunk 16]
```diff
@@ -146,7 +146,6 @@ pub struct VMState {
 // Totally arbitrary, probably we will have to change it later.
 pub const DEFAULT_INITIAL_GAS: u32 = 1 << 16;
 impl VMState {
-    // TODO: The VM will probably not take the program to execute as a parameter later on.
```
Comment (JulianVentura) [inline src/state.rs:149]: Created issue #116

**Pull request merged**